### ISSUE TITLE
Does this library support targeted classification in its current state?

### ISSUE THREAD

**Issue by MackBlackburn** [QData/TextAttack#732]
Do any of the attacks support targeted classification as a goal function? I do not see it stated anywhere but untargeted classification is mentioned. Manually switching the goal function of a recipe results in an error:

`AttributeError: 'UntargetedClassification' object has no attribute 'num_queries'`

Here is some basic code I am using that causes the error:
```

class HuggingFaceSentimentAnalysisPipelineWrapper(ModelWrapper):
    def __init__(self, model):
        self.model = model


    def __call__(self, text_inputs):
        raw_outputs = self.model(text_inputs)
        outputs = []
        for sub_l in raw_outputs:
            out_scores = [0] * len(self.model.model.config.label2id)
            for d in sub_l:
                score = d["score"]
                out_scores[self.model.model.config.label2id[d['label']]] = score
            outputs.append(out_scores)
        return np.array(outputs)

pipe = pipeline("text-classification", 'j-hartmann/emotion-english-distilroberta-base', device=0, return_all_scores=True)

model_wrapper = HuggingFaceSentimentAnalysisPipelineWrapper(pipe)
recipe = TextBuggerLi2018.build(model_wrapper)
recipe.transformation.language = "eng"
recipe.constraints = []
recipe.goal_function = TargetedClassification(model_wrapper, target_class=2)
recipe.goal_function.num_queries = 10

dataset = Dataset([('Im angry', 0)])
attacker = Attacker(recipe, dataset)
results = attacker.attack_dataset()

```

**Comment by qiyanjun** [comment]
Yes, targeted classification is fully supported — `TargetedClassification` is a real, working goal function (used by e.g. `InputReductionFeng2018`), it just needs to be *instantiated*, not attribute-swapped onto an existing `Attack`:

\`\`\`python
from textattack.goal_functions.classification import TargetedClassification

goal_function = TargetedClassification(model_wrapper, target_class=2)  # attack targets class 2
attack = Attack(goal_function, constraints, transformation, search_method)
\`\`\`

The `AttributeError: 'UntargetedClassification' object has no attribute 'num_queries'` you hit suggests whatever swap you did bypassed `__init__` (where `num_queries` gets set) — e.g. assigning the class itself rather than calling it, or manually poking at `attack.goal_function` after construction. If you share the exact code that produced that error I can pin down precisely what went wrong, but the library-level answer to your question is yes, this is supported.

**Comment by qiyanjun** [comment]
Answered above — targeted classification is fully supported via `TargetedClassification`, with a working example. Closing as answered; reopen if the example doesn't resolve your case.
